src/components/ReportsPage/HeatmapPage/HeatmapUtils.ts

```diff
@@ -94,25 +116,147 @@ const groupReportDataBy = <T extends GroupByValues>(
 export const getScorecardServiceScoresByGroupByOption = (
   scores: ScorecardServiceScore[] | undefined,
   groupBy: GroupByOption | undefined,
+  domainTagByEntityId: StringIndexable<string[]>,
 ): StringIndexable<ScorecardServiceScore[]> => {
   if (scores === undefined || scores.length === 0) {
     return {};
   }
 
   switch (groupBy) {
-    case GroupByOption.SERVICE:
+    case GroupByOption.ENTITY:
       return _groupBy(scores, (s: ScorecardServiceScore) => s.serviceId);
     case GroupByOption.SERVICE_GROUP:
       return groupReportDataBy(scores, 'tags');
     case GroupByOption.TEAM:
       return groupReportDataBy(scores, 'teams');
     case GroupByOption.LEVEL:
       return groupReportDataBy(scores, 'ladderLevels');
+    case GroupByOption.DOMAIN:
+      return groupReportDataBy(scores, 'domains', domainTagByEntityId);
     default:
       return {};
   }
 };
 
+
+interface HierarchyNode {
+  node: {
+    tag: string;
+  }
+  orderedChildren: HierarchyNode[];
+}
+
+export const hierarchyNodeFlatChildren = (node: HierarchyNode): string[] => {
+  return node.orderedChildren.flatMap((child) => [child.node.tag, ...hierarchyNodeFlatChildren(child)]);
+}
+
+export const hierarchyNodeFlatAllChildren = (node: HierarchyNode[], result: Record<string, string[]> = {}): Record<string, string[]> => {
```

Comment:
Why does this take a 2nd param to assign if it's unused but then return that value? It would be clearer and more consistent if we just initialized the var
```suggestion
export const hierarchyNodeFlatAllChildren = (node: HierarchyNode[]): Record<string, string[]> => {
  const result: Record<string, string[]> = {};
```

Reply:
This method was called recursively with the parameter, removed anyway when changing the domains stuff.